Telerik.Sitefinity.Amazon/BlobStorage/AmazonBlobStorageProvider.cs

```diff
@@ -26,25 +26,42 @@ public class AmazonBlobStorageProvider : CloudBlobStorageProvider
         /// <param name="config">The collection of parameters (each by its name and value) of the current provider's configuration settings.</param>
         protected override void InitializeStorage(NameValueCollection config)
         {
+            var useIamInstanceRoleValue = "false";
+            if (config.AllKeys.Contains(UseIamInstanceRoleKey))
+                useIamInstanceRoleValue = config[UseIamInstanceRoleKey].Trim();
+
+            if (!bool.TryParse(useIamInstanceRoleValue, out this.useIamInstanceRole))
+                throw new ConfigurationException("'{0}' unable to parse {1}".Arrange(UseIamInstanceRoleKey, useIamInstanceRoleValue));
+
             this.accessKeyId = config[AccessKeyIdKey].Trim();
-            if (String.IsNullOrEmpty(this.accessKeyId))
+            if (!this.useIamInstanceRole && String.IsNullOrEmpty(this.accessKeyId))
                 throw new ConfigurationException("'{0}' is required.".Arrange(AccessKeyIdKey));
 
             this.secretKey = config[SecretKeyKey].Trim();
-            if (String.IsNullOrEmpty(this.secretKey))
+            if (!this.useIamInstanceRole && String.IsNullOrEmpty(this.secretKey))
                 throw new ConfigurationException("'{0}' is required.".Arrange(SecretKeyKey));
 
             this.bucketName = config[BucketNameKey].Trim();
             if (String.IsNullOrEmpty(this.bucketName))
                 throw new ConfigurationException("'{0}' is required.".Arrange(BucketNameKey));
 
+            if (config.AllKeys.Contains(KeyPrefixKey))
```

Comment:
Optional configuration item to allow the use of a key prefix on files uploaded to the S3 Bucket